### ISSUE TITLE
Modularization

### ISSUE THREAD

**Issue by JaKXz** [h2non/jshashes#31]
This is a great package, and thank you so much for it! However, I think it could be improved if each algorithm was it's own file, for those of us who only need one or two different algorithms.

Then, the build process could concatenate all the files and your `bower.json` could remain the same, but overrides could be specified as necessary, e.g.

```json
"overrides": {
  "jshashes": {
    "main": "sha256.js"
  }
}
```

What do you think?

**Comment by h2non** [comment]
I tend to agree. This was clearly an smell when the library borns. Isolation is generally a good approach, It's more maintainable, more readable and hygienic.

From the other hand, taking into account the nature of node's modules, loading a bunch of them is not a problem, neither in the performance side or bad practice. IMO the final benefit of doing this is not too much considerable.

To be honest, I don't have in mind releasing a fresh version which splits the algorithms, but maybe I'll be able to it in a future, however I'm always open to accept a PR, specially for this kind of features.

The hard part here is about coupling, since some of them are reused between multiple algorithms, so you should be careful.

For browser packaging, my proposal would be using browserify.
